src/tx/applyTx/checkDeposit.js

```diff
@@ -35,8 +37,9 @@ module.exports = (state, tx, bridgeState) => {
   if (
     !deposit ||
     !equal(BigInt(deposit.amount), BigInt(tx.outputs[0].value)) ||
-    Number(deposit.color) !== tx.outputs[0].color ||
-    !addrCmp(deposit.depositor, tx.outputs[0].address)
+    Number(deposit.color) !== txColor ||
+    !addrCmp(deposit.depositor, tx.outputs[0].address) ||
+    deposit.data !== tx.outputs[0].data
```

Comment:
what about NFT deposits that don't have data. will the comparison of `null !== null` work?

Reply:
Good catch 🐰